controller/gke-cluster-config-handler.go

```diff
@@ -283,7 +283,9 @@ func (h *Handler) enqueueUpdate(config *gkev1.GKEClusterConfig) (*gkev1.GKEClust
 		h.gkeEnqueue(config.Namespace, config.Name)
 		return config, nil
 	}
+	oldName := config.Name
 	config = config.DeepCopy()
+	config.Name = oldName
```

Comment:
Do we have any idea why the name isn't being copied?

Reply:
Yes, and does this issue also exist in EKS ?

Reply:
I am wrong on this. The real problem is when `enqueueUpdate` tries to update status it consistently runs into a conflict here

https://github.com/rancher/gke-operator/blob/master/controller/gke-cluster-config-handler.go#L288

`UpdateStatus` creates an empty spec object and when it returns it returns the empty result, which has `""` set as the Name.

So it's not the deepcopy, the real fix will be more complicated and isn't central to the labels issue, removing for now. It is probably also a problem for EKS.